libraries/chain/include/eosio/chain/pending_snapshot.hpp

```diff
@@ -33,9 +33,16 @@ class pending_snapshot {
       return snapshots_dir / fc::format_string(".incomplete-snapshot-${id}.bin", fc::mutable_variant_object()("id", block_id));
    }
 
-   T finalize(const chain::controller& chain) const {
+   T finalize(const block_id_type& lib_id, const chain::controller& chain) const {
+      auto lib_num = chain::block_header::num_from_id(lib_id);
+      auto block_num = chain::block_header::num_from_id(block_id);
+
+      EOS_ASSERT(lib_num >= block_num, chain::snapshot_finalization_exception,
```

Comment:
How does this ever occur -- why not assert()?

Reply:
It could be an assert() I believe. I'll update.